src/github3/repos/branch.py

```diff
@@ -74,62 +74,64 @@ def protection(self):
         return BranchProtection(json, self)
 
     @decorators.requires_auth
-    def protect(self, enforcement=None, status_checks=None):
-        """Enable force push protection and configure status check enforcement.
+    def protect(self, enforce_admins, required_pull_request_reviews,
+                required_status_checks, restrictions):
+        """Enable force push protection.
 
-        See: http://git.io/v4Gvu
+        See:
+        https://developer.github.com/v3/repos/branches/#update-branch-protection
 
-        :param str enforcement:
-            (optional), Specifies the enforcement level of the status checks.
-            Must be one of 'off', 'non_admins', or 'everyone'. Use `None` or
-            omit to use the already associated value.
-        :param list status_checks:
-            (optional), An list of strings naming status checks that must pass
-            before merging. Use `None` or omit to use the already associated
-            value.
+        :param enforce_admins:
+            TODO
+        :type enforce_admins:
+            bool
```

Comment:
For simple types, I'd prefer to put these in-line, e.g., `:param bool enforce_admins`